RecoTracker/TkSeedGenerator/plugins/DeepCoreSeedGenerator.h

```diff
@@ -0,0 +1,172 @@
+
+#include "TrackingTools/GeomPropagators/interface/StraightLinePlaneCrossing.h"
+#include "TrackingTools/GeomPropagators/interface/Propagator.h"
+#include "TrackingTools/Records/interface/TrackingComponentsRecord.h"
+
+#include "MagneticField/Engine/interface/MagneticField.h"
+#include "MagneticField/Records/interface/IdealMagneticFieldRecord.h"
+
+#include <boost/range.hpp>
+#include <boost/foreach.hpp>
+#include "boost/multi_array.hpp"
+
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "CommonTools/UtilAlgos/interface/TFileService.h"
+
+#include "SimDataFormats/Vertex/interface/SimVertex.h"
+
+#include "Geometry/CommonDetUnit/interface/PixelGeomDetUnit.h"
+
+#include "DataFormats/TrajectorySeed/interface/TrajectorySeedCollection.h"
+
+#include "TTree.h"
+#include "PhysicsTools/TensorFlow/interface/TensorFlow.h"
+
+namespace edm {
+  class Event;
+  class EventSetup;
+}  // namespace edm
+
+class DeepCoreSeedGenerator : public edm::stream::EDProducer<> {
+public:
+  explicit DeepCoreSeedGenerator(const edm::ParameterSet&);
+  ~DeepCoreSeedGenerator() override;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+  // A pointer to a cluster and a list of tracks on it
+  struct TrackAndState {
+    TrackAndState(const reco::Track* aTrack, TrajectoryStateOnSurface aState) : track(aTrack), state(aState) {}
+    const reco::Track* track;
+    TrajectoryStateOnSurface state;
+  };
+
+  template <typename Cluster>
+  struct ClusterWithTracks {
+    ClusterWithTracks(const Cluster& c) : cluster(&c) {}
+    const Cluster* cluster;
+    std::vector<TrackAndState> tracks;
+  };
+
+  typedef ClusterWithTracks<SiPixelCluster> SiPixelClusterWithTracks;
+
+  typedef boost::sub_range<std::vector<SiPixelClusterWithTracks>> SiPixelClustersWithTracks;
+
+  double jet_pt;
+  double jet_eta;
+  double pitchX = 0.01;           //100 um (pixel pitch in X)
+  double pitchY = 0.015;          //150 um (pixel pitch in Y)
+  static const int jetDimX = 30;  //pixel dimension of NN window on layer2
+  static const int jetDimY = 30;  //pixel dimension of NN window on layer2
+  static const int Nlayer = 4;    //Number of layer used in DeepCore
+  static const int Nover = 3;     //Max number of tracks recorded per pixel
+  static const int Npar = 5;      //Number of track parameter
+
+private:
+  void beginJob();
+  void produce(edm::Event&, const edm::EventSetup&) override;
+  void endJob();
+
+  // ----------member data ---------------------------
+  std::string propagatorName_;
+  edm::ESHandle<MagneticField> magfield_;
+  edm::ESHandle<GlobalTrackingGeometry> geometry_;
+  edm::ESHandle<Propagator> propagator_;
+
+  edm::EDGetTokenT<std::vector<reco::Vertex>> vertices_;
+  edm::EDGetTokenT<edmNew::DetSetVector<SiPixelCluster>> pixelClusters_;
+  std::vector<SiPixelClusterWithTracks> allSiPixelClusters;
+  std::map<uint32_t, SiPixelClustersWithTracks> siPixelDetsWithClusters;
```

Comment:
Seems unused?

Reply:
These fours are used in the .cc in several points. Is this usage incorrect?

Reply:
ok no, you mean only the last one, yes it can be removed